### ISSUE TITLE
mistake

### ISSUE THREAD

**Issue by sugar-chen-xq** [vsangelidakis/CLUMP#1]
https://github.com/vsangelidakis/CLUMP/blob/a62641a6413c65dbc146bbf9310cf8719f38c097/functions/ExtractSurface.m#L113
 
Thank you very much for your excellent work. There is a minor mistake.
alph=acos( (r1^1+d^2-r2^2) / (2*r1*d) ); 
↓
alph=acos( (r1^2+d^2-r2^2) / (2*r1*d) );

**Comment by vsangelidakis** [comment]
Hi @sugar-chen-xq . Thanks for sharing this! Looking into the [calculations for the intersection of two spheres](https://mathworld.wolfram.com/Sphere-SphereIntersection.html), your comment seems correct, but when I tried it it gave me bad surface reconstructions when I tested this on a complex case. I need to test this in more detail in the coming days. Have you tested this on your clumps?

**Comment by sugar-chen-xq** [comment]
Hi @vsangelidakis . This problem did bother me for a long time.
1. I tested the **ExtractSurface.m**  in modules and found no error.
2. Then I noticed that in the subfunction **ExtractSurface.m**, the **contact detection** module is located before **the dispersion of the intersection of two spheres**.

> I think this is the immediate cause of bad surface reconstructions. The _circle vertices_, which is part of the sphere, should be integrated into the _spherical vertices_ before deleting the internal points.

3. The modified code performs well in the new tests for complex clumps.


[corrected-ExtractSurface.zip](https://github.com/vsangelidakis/CLUMP/files/8518151/corrected-ExtractSurface.zip)

**Comment by sugar-chen-xq** [comment]
![test](https://user-images.githubusercontent.com/58979758/164149046-29b54b20-dd24-4ac6-81c8-61b479c774ec.JPG)

**Comment by vsangelidakis** [comment]
Hi @sugar-chen-xq - You are absolutely right!

I tested your modified function script and it indeed works for some complex cases where the existing script failed to reconstruct the surface correctly. You actually found 2 bugs, one in the calculation of the distance and one where the inner points were not deleted after the addition of the intersection points.

I have updated the source code according to your correction - Thank you very much! Feel free to send pull requests if you would like to propose more improvements down the road.
